### ISSUE TITLE
fix: Cannot read property 'add' of undefined  at onGlobalSetup

### ISSUE THREAD

**Issue by JoeyYoung1997** [nuxt-community/composition-api#488]
**🐛 The bug**
What isn't working? Describe what the bug is.
I found out after debugger, "const onGlobalSetup =(fn)=> {globalSetup.add(fn);};", in this code, globalSetup has not yet been assigned in the globalPlugin function

code show as below:

export default ({ app, store }, inject) => {
  onGlobalSetup(() => {
    inject('navigateTo', func)
  })
}

**Comment by danielroe** [comment]
@JoeyYoung1997 Would you provide a reproduction? 🙏 

[Working example](https://codesandbox.io/s/nifty-frost-bf0in?file=/plugins/setup.js).

**Comment by negibouze** [comment]
I got the same error.
The following error is displayed on the browser console.

Chrome:

```
TypeError: Cannot read property 'add' of undefined
    at onGlobalSetup (index.mjs:459)
```

Firefox:

```
TypeError: globalSetup is undefined
    onGlobalSetup index.mjs:459
```

**Comment by josendf-basetis** [comment]
Same issue here, 

````
client.js:102 TypeError: Cannot read property 'add' of undefined
    at onGlobalSetup (index.mjs:459)
````

While calling `onGlobalSetup` during a client plugin registration, seems that the variable `globalSetup`  in the `index.msj` module has not been initialized:

```
let globalSetup;
const onGlobalSetup = (fn) => {
  globalSetup.add(fn);
};
```

**Comment by josendf-basetis** [comment]
Hi @danielroe, 

Thank you for your prompt response, following is a reproduction:

https://codesandbox.io/s/goofy-night-t2sr6

The error occurs in `plugin/someplugin.client.js`

```
TypeError: Cannot read property 'add' of undefined
```

**Comment by tobiasdiez** [comment]
Calling `globalPlugin` before works for me as a workaround:
```js
import { provide, onGlobalSetup, globalPlugin } from '@nuxtjs/composition-api'

const myPlugin: Plugin = function (context, inject) {
  globalPlugin(context, inject)
  onGlobalSetup(() => {
    provide(...)
  })
}
```

**Comment by danielroe** [comment]
This should be fixed in `0.24.1` 😊
